app/models/genre.rb

```diff
@@ -0,0 +1,15 @@
+class Genre < ApplicationRecord
+  has_and_belongs_to_many :books
+
+  validates :name,
+            presence: true,
+            uniqueness: true
```

Comment:
Some further information about uniqueness validations for you to know -

This validation can't really prevent duplicates from going into the system. If someone was on the create Genre form and they pressed the submit button to submit their genre 10 times rapidly, it will spawn off 10 requests to create the Genre.

It's likely up to 10 of those will go through and at the time of the Genre checking whether its name is unique, there will be one of the preceding Genres on its way into the DB. Therefore the validation will pass, because at that exact moment in time there is no Genre with the same name in the DB.

With these race conditions, of the 10 requests you could end up with maybe 2-3 Genres with the same name in the DB.

This isn't going to be a common occurrence, but it CAN happen.

Something else to know about uniqueness validations is that they are really slow.

Under the hood they basically do something like `Genre.exists?(name: name)`. For a large table that can take 10s/100s of milliseconds, and will happen for every save of every Genre record. The longer it takes, the more time we have for multiple concurrent requests to put "invalid" data in the database.

So - should we really use uniqueness validations in real production apps? Not really - not if we can afford not to.

But - this is all next-level™ knowledge to go through in more detail a little later.

For now - we can leave this validation here but let's also add a database-level validation that CAN well and truly prevent duplicates.

You'll need to add a unique index to your name field.

In your migration, that'll look something like:

`add_index :genres, :name, unique: true`

In the scenario outlined at the top of this comment, the end user would instead see a 500 error on their form.

What would happen is:

1. Request 1 is submitted
2. Request 2 is submitted a few ms later
3. Request 1 commits Genre with name "History" to the DB
4. Concurrently, Request 2 is checking whether a Genre called "History" is in the DB. There isn't at this exact moment in time

Scenario: Validation and no DB index:
5. Request 2 persists a second Genre called "History" to the DB a few ms after Request 1 done

Scenario: Validation and DB index:
6. Request 2 attempts to persist a second Genre called "History" to the DB a few ms after Request 1 done. The DB transaction to do this fails and rolls back, because the Genre from Request 1 is in the DB and this violates the DB index that was created.
7. The end user sees a 500 error on screen. In the fullness of time, you'd teach your model how to listen for unique index exceptions coming from the DB, and handle them gracefully so instead the user would see a validation error like "name is already taken"